File: MovieApp/DTOs/PagedResultDto.cs
namespace MovieApp.DTOs;

public class PagedResultDto<T>
{
    public List<T> Items { get; set; } = new();
    public int TotalCount { get; set; }
    public int PageSize { get; set; }
    public int CurrentPage { get; set; }

    public int TotalPages => 
        (int)Math.Ceiling((double)TotalCount / PageSize);
}

File: movie-app-frontend/src/app/models/dtos/PagedResultDto.ts
export class PagedResultDto<T> {
  items: T[] = [];
  totalCount: number = 0;
  pageSize: number = 10;
  currentPage: number = 1;
  totalPages: number = 0;

  constructor(init?: Partial<PagedResultDto<T>>) {
    Object.assign(this, init);
  }
}